Tente de créer un raccourcis pour que les modo puissent supprimer les mauvais domaines

artragis opened the pull request:
Je fais une petite PR pour faciliter le ban des domaines de fournisseurs de membres qui ne se connectent que pou spam.
Aujourd'hui une fois le ban fait, il faut aller sur la page des fournisseur, vérifier que le membre qu'on a banni n'est pas en haut de la liste et bannir si c'est nécessaire.

L'ambition (petite) de cette PR est d'ajouter le bouton directement dans l'interface de profile une fois le membre banni.
Mon environnement étant dans les choux, je n'ai pas pu tester, d'où le brouillon, cependant le code devrait être prêt à être reviewé.

artragis committed:
Tente de créer un raccourcis pour que les modo puissent supprimer les mauvais domaines

Arnaud-D commented on templates/member/profile.html:
```diff
@@ -856,6 +856,14 @@ <h3>{% trans 'Sanctions' %}</h3>
                             </form>
+                                <!-- if the user was banned and he was the first for his mail provider, propose
+                                quick access to ban provider -->
+                                {% if profile.user | has_new_provider and can_ban_provider %}
```
Je pense que ça serait mieux de remonter cette logique dans le code et pas dans le template. Dans le template, on aurait juste quelque chose du genre `{% if show_ban_provider_button %}`.

Arnaud-D commented on templates/member/profile.html:
```diff
@@ -856,6 +856,14 @@ <h3>{% trans 'Sanctions' %}</h3>
+                                <!-- if the user was banned and he was the first for his mail provider, propose
+                                quick access to ban provider -->
+                                {% if profile.user | has_new_provider and can_ban_provider %}
+                                    <form method="post" action="{% url 'check-new-email-provider' user|get_new_provider_pk %}">
```
Il y a vraiment besoin d'un filtre ? Des fois qu'il y aurait une méthode (éventuellement à créer), on pourrait faire `profile.get_new_provider` ou quelque chose du genre. Ça nécessiterait d'utiliser le profil à la place du `user` par contre.

artragis committed:
prend en compte les retours

philippemilink committed:
Merge branch 'dev' into feature/quick_domain_ban

philippemilink committed:
Corrige le nom d'une fonction

philippemilink committed:
Corrections mineures

- corrige le style de la popup
- supprime l'ajout d'un import inutile
- fait la requête pour savoir s'il faut bannir le domaine mail seulement
  si on a le droit de le bannir

philippemilink committed:
Merge branch 'dev' into feature/quick_domain_ban

philippemilink committed:
Merge branch 'dev' into feature/quick_domain_ban

philippemilink committed:
Merge branch 'dev' into feature/quick_domain_ban

Arnaud-D commented on templates/member/profile.html:
```diff
@@ -858,10 +858,16 @@ <h3>{% trans 'Sanctions' %}</h3>
+                                    <form method="post" action="{% url 'check-new-email-provider' provider.pk %}" id="ban-provider-{{ provider.pk }}"
+                                    class="modal modal-flex">
+                                    <p>
+                                        {% trans "Bannir le fournisseur email " %} {{ provider.provider }} ?
```
Cela devrait être un blocktrans pour être propre niveau traductibiblité. Mais pas bloquant parce qu'on n'a pas prévu de traduire de si tôt...

philippemilink replied:
C'est corrigé.

Arnaud-D commented:
À part ma remarque mineure ci-dessus, c'est bon.

philippemilink committed:
Merge branch 'dev' into feature/quick_domain_ban

philippemilink committed:
Utilise un blocktrans au lieu de {% trans "foo" %}

philippemilink commented:
Instructions pour la QA : 
1. inscrire un nouvel utilisateur avec un nom de domaine pour l'email pas encore connu du site
2. activer le compte en utilisant l'adresse envoyée par mail, visible dans la console
3. se connecter en tant qu'admin, bannir définitivement le membre qui vient de s'inscrire
4. tester que la popup pour bannir le fournisseur d'email fonctionne bien

Arnaud-D reviewed (dismissed):
QA NOK

En fait, j'ai une réserve sur l'affichage. On ne peut pas tester facilement en local avec des droits réalistes. Tous les staff sont admin ou presque sur l'environnement de dév. Je pense que ça s'affiche pour tous les staff, alors que les modérateurs en général n'ont pas la gestion des emails (je ne l'avais pas à l'époque où j'étais staff, de mémoire).

Arnaud-D reviewed (changes requested):
QA NOK

En fait, j'ai une réserve sur l'affichage. On ne peut pas tester facilement en local avec des droits réalistes. Tous les staff sont admin ou presque sur l'environnement de dév. Je pense que ça s'affiche pour tous les staff, alors que les modérateurs en général n'ont pas la gestion des emails (je ne l'avais pas à l'époque où j'étais staff, de mémoire).

C'est pas bloquant, juste des staff qui auront un bouton qui ne marche pas.

philippemilink commented:
Euh non, puisque `provider_to_ban` n'est défini que si on le droit de gérer les mails. Si on n'a pas le droit, `provider_to_ban` est évalué à `None`/`False` dans le template, et le bouton ne s'affiche pas.
